### PR TITLE
chore: install react navigation and imports sort plugin

### PR DESCRIPTION
**Pull request by Stukz**
### Description
This PR includes:

1. installation of React Navigation and initial setup
2. installation of  [prettier-plugin-sort-imports](https://github.com/trivago/prettier-plugin-sort-imports) to define a standard order for the imports

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
   bracketSameLine: true,
-  bracketSpacing: false,
+  bracketSpacing: true,
   singleQuote: true,
   trailingComma: 'all',
+  printWidth: 100,
+  jsxBracketSameLine: true,
+  importOrder: ['^@(.*)/(.*)$', '^screens/(.*)$', '^constants/(.*)$', '^[./*]'],
+  importOrderSeparation: true,
+  importOrderSortSpecifiers: true,
```
Comment (aguscha333) [inline .prettierrc.js:11]: Nice!

File: src/screens/WelcomeScreen/index.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
```
Comment (aguscha333) [inline src/screens/WelcomeScreen/index.tsx:1]: I like more putting an index file with
```
export { default } from './HomeScreen'
```
and then having `HomeScreen.tsx` and `HomeScreen.styles.ts`

So there is not a hundred files named the same, it makes files easier to find. WDYT?

File: src/screens/WelcomeScreen/index.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  DebugInstructions,
+  Header,
+  LearnMoreLinks,
+  ReloadInstructions,
+} from 'react-native/Libraries/NewAppScreen';
+
+import styles from './styles';
+
+const Section: React.FC<{
+  title: string;
```
Comment (asdolo) [inline src/screens/WelcomeScreen/index.tsx:14]: What do you thing about separating the interface for the props?

```ts
interface SectionProps {
  title: string;
}; 

const Section: React.FC<SectionProps> = ({ children, title }) => {

  ...

}
```

File: src/screens/WelcomeScreen/index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  Colors,
+  DebugInstructions,
+  Header,
+  LearnMoreLinks,
+  ReloadInstructions,
+} from 'react-native/Libraries/NewAppScreen';
+
+import styles from './styles';
+
+const Section: React.FC<{
```
Comment (asdolo) [inline src/screens/WelcomeScreen/index.tsx:13]: What's your viewpoint on declaring more than one React Component per file?

- Is it worth to move this to a separate file?
- Is it worth to do it now? Or maybe only when it's needed?

**Review by asdolo — APPROVED** [review]

File: src/screens/WelcomeScreen/index.tsx (hunk 2 above)
Comment (aguscha333) [reply src/screens/WelcomeScreen/index.tsx:1]: yes and also it is easier to find component files if they are not all named index

Comment (asdolo) [reply src/screens/WelcomeScreen/index.tsx:1]: Right. On the other side we'll have to create one more file for every new component that is redundant 🤔.

File: src/screens/WelcomeScreen/index.tsx (hunk 3 above)
Comment (asdolo) [reply src/screens/WelcomeScreen/index.tsx:14]: Yes, I was thinking of this as a good practice for all of us to have, and also for new people reading TypeScript code.

File: src/screens/WelcomeScreen/index.tsx (hunk 2 above)
Comment (aguscha333) [reply src/screens/WelcomeScreen/index.tsx:1]: Yup, I agree that neither solution is perfect but having worked with both I prefer much more the one that I am proposing, I think it also scales better in the developer experience aspect.
We can leave it up to the developer as well, if the developer is opinionated enough on the matter they will have no problem changing it. If they aren't they will use what's in the base.

File: src/screens/WelcomeScreen/index.tsx (hunk 3 above)
Comment (aguscha333) [reply src/screens/WelcomeScreen/index.tsx:14]: yuppp totally agree on that, just not sure if the change in worth in this instance

Comment (Stukz) [reply src/screens/WelcomeScreen/index.tsx:14]: yes It was the boilerplate from RN CLI, just to have something to render

File: src/screens/WelcomeScreen/index.tsx (hunk 2 above)
Comment (Stukz) [reply src/screens/WelcomeScreen/index.tsx:1]: maybe we could bring this discussion on the slack channel and use a poll

**Review by aguscha333 — APPROVED** [review]

**Review by jpirazusta — APPROVED** [review]

**Pull request merged**